core/src/main/java/de/jplag/Submission.java

```diff
@@ -309,4 +311,21 @@ public Submission copy() {
         copy.setBaseCodeComparison(baseCodeComparison);
         return copy;
     }
+
+    /**
+     * @return A mapping of each file in the submission to the number of tokens in the file
+     */
+    public Map<File, Integer> getTokenCountPerFile() {
+        if (this.tokenList == null) {
+            return Collections.emptyMap();
+        }
+        Map<File, Integer> tokenCounts = new HashMap<>();
+        for (File file : this.files) {
+            tokenCounts.put(file, 0);
+        }
+        for (Token token : this.tokenList) {
+            tokenCounts.put(token.getFile(), tokenCounts.get(token.getFile()) + 1);
+        }
+        return tokenCounts;
```

Comment:
This is done every time this method is called. For API users, it would be better if the map is stored as field, and calculated upon the first call of the method. As in: If empty, calculate, if not, return.